libs/math/include/math/distance/absolute.hpp

```diff
@@ -30,31 +30,31 @@ typename ArrayType::Type AbsoluteDistance(ArrayType const &A, ArrayType const &B
   auto it1   = A.cbegin();
   auto it2   = B.cbegin();
   assert(it1.size() == it2.size());
-  auto ret = Type(0);
+  auto ret = Type{0};
   Type d;
 
   while (it1.is_valid())
   {
-    if ((*it1 > 0) && (*it2 > 0))
+    if ((*it1 > Type{0}) && (*it2 > Type{0}))
```

Comment:
As Type{0} is used in many places here, may be 
static Type const ZERO = Type{0}; 
and use it everywhere?